### ISSUE TITLE
Actions Court Prep Worksheet

### ISSUE THREAD

**Issue by gretchent74** [MN-Script-Team/DHS-PRISM-Scripts#594]
Is any one getting this error:  Script Error at line 1, Column1: Type mismatch: 'read_PALC_last_payment_date"?

**Comment by gretchent74** [comment]
@MN-Script-Team/prism-scriptwriters anybody?????

**Comment by paschell** [comment]
Hi @gretchent74 !   I get the Script Error at Line1, Column 1: mismatch : 'go_to'

**Comment by theVKC** [comment]
Anyone else?

**Comment by KallistaImdieke** [comment]
I am getting the same error here in Stearns County

**Comment by heatherallen** [comment]
I am getting the same error here in Scott County "Script Error at Line1, Column 1: mismatch : 'go_to'"

**Comment by theVKC** [comment]
OK, so it definitely does, in fact, look like a bug. Any chance we can get a quick resolution on it from anyone?

**Comment by RoberT-1000** [comment]
The fix is already in the MASTER branch. I removed the function `go_to` with `write_value_and_transmit` but we were waiting on something. (See #575 )

**Comment by RoberT-1000** [comment]
Also see #579

**Comment by theVKC** [comment]
!!!

I see now! Because master branch has the new file locations the redirects are pointing toward release.

@RoberT-1000: care to pull the same fix into the release version?

**Comment by theVKC** [comment]
Yes, all, @RoberT-1000 fixed this in #579. If @RoberT-1000 would be so kind as to pull it into release as well, we should be square.

**Comment by RoberT-1000** [comment]
Will do. Is there anything else I should do on this?

**Comment by RoberT-1000** [comment]
Done

**Comment by theVKC** [comment]
Nah, you should be square. Thank you so much @RoberT-1000!

**Comment by gretchent74** [comment]
@RoberT-1000 You are Awesome!

**Comment by RoberT-1000** [comment]
I'll accept "helpful" for this one...that's a pretty low bar for "awesome" :)

**Comment by theVKC** [comment]
Reopening because the issue is `read_PALC_last_payment_date` and Gretchen is still seeing it. I'm on with her via WebEx right now and we'll work this issue together.

**Comment by theVKC** [comment]
@RoberT-1000, is there any chance we can confirm that court-prep-worksheet is working as expected? I'm starting to think I kept this open for too long.

**Comment by RoberT-1000** [comment]
Now I'm getting a new error message

![image](https://cloud.githubusercontent.com/assets/22157633/20348558/ac91751a-abcb-11e6-844a-67a7e61a37e5.png)

**Comment by RoberT-1000** [comment]
Which, now that I dig into the FuncLib, is a function that doesn't exist. @levesswe >> thoughts?

**Comment by RoberT-1000** [comment]
I'm looking in to it

**Comment by RoberT-1000** [comment]
OK, so I ran it locally, and it runs just fine. I think it's a problem of the scripts falling too deep down the rabbit hole. The function `read_PALC_last_payment_date` is local to that script and it is called within other functions in the script. I've seen this with other scripts that call functions from within functions. Because we are essentially calling a function to call the redirect which then calls a function to load the script which then calls the functions library and then runs the script which then calls a function to call a function...the script gets a little bit of digital vertigo. 

My recommendation treatment, doctors, is to lobotomize the function `read_PALC_last_payment_date` and replace the function calls with the functionality from the function...especially since the function is only used four times and they are all within the function `create_PALC_variable`...

![image](https://cloud.githubusercontent.com/assets/22157633/20349100/762f46b6-abce-11e6-853d-cbb793f2208b.png)

**Comment by RoberT-1000** [comment]
Correction, it's only being called once...I'm waiting for the coffee to kick in...

**Comment by theVKC** [comment]
@RoberT-1000: thank you for investigating! Can you hit it right quick so we can resolve it?

**Comment by RoberT-1000** [comment]
On it! I was at a Gen Y training this morning so I could learn how to effectively communicate with myself...

**Comment by RoberT-1000** [comment]
My concern...and I cannot verify until the script goes lives on GH, is that the next couple functions will suffer the same fate. Because `read_PALC_payment_type`, `read_PALC_payment_amount` and `read_PALC_alloc_amount` are ONLY used in this one script in this one function, I am going to replace those as well...because future future proofing...sorry @levesswe

**Comment by levesswe** [comment]
Robert, do what you need to do with my complete approval and satisfaction.  I would much rather see this script working for the collaborative than worry about how it does what it does.

**Comment by theVKC** [comment]
Beautiful. Thank you!

**Comment by theVKC** [comment]
Reopening for testing. @heatherallen is testing, I'll see if someone here can also test it.

**Comment by theVKC** [comment]
@khiestand has confirmed that this is now working through here, but it threw a Word-doc related bug. @RoberT-1000, any chance the document is defined locally instead of with global variables?

**Comment by heatherallen** [comment]
@theVKC I got the change log pop this time.  I'm receiving a Word-doc related bug as well where the word doc opens but then get the error "script error at line 1, column 1, this filed could not be found."

**Comment by KallistaImdieke** [comment]
I am getting the following error when i try and run this 

![capture](https://cloud.githubusercontent.com/assets/19214103/20392509/c7283882-ac9d-11e6-8e7e-d5055ba7a87a.PNG)

**Comment by RoberT-1000** [comment]
I know what that is...sorry

**Comment by RoberT-1000** [comment]
Nevermind...I thought I had left the globvar call at the top of the script.

**Comment by theVKC** [comment]
I'm going to take a wild guess and say that there's a Word doc you all don't have installed locally.

@RoberT-1000: can you email "Court Prep Worksheet.dotx" to the group? If you save that in the folder defined by the variable word_documents_folder_path, this should work for you as well as it works in Anoka.

In any case, this particular issue (the PALC function deal) appears to be resolved. Closing.

**Comment by theVKC** [comment]
Also on that last note: https://github.com/MN-Script-Team/BZS-FuncLib/issues/234 will likely resolve this, so for now, no need to create a new issue. Hoping to take care of that today.
